File: sdk/js-client-sdk/src/transport/transport.ts
import { Mode, IContext } from "../context";
import Streamer from "./streamer";
import Poller from "./poller";
import { EventProducer } from "../events";

export interface ITransport {
  start: () => void;
  stop: () => void;
}

export default function Transport(clientKey: string, context: IContext, events: EventProducer): ITransport {
  return context.getConfig().mode === Mode.STREAMING
    ? Streamer(clientKey, context, events)
    : Poller(clientKey, context, events);
}

File: sdk/python-server-sdk/flagbase/transport/transport.py
from flagbase.config import Mode
from flagbase.context import Context
from flagbase.events import Events
from .poller import Poller

class Transport:
    def __init__(self, context: Context, events: Events):
        self.context = context
        self.events = events

        config = context.get_config()
        if config.mode == Mode.STREAMING:
            raise ValueError(f"{Mode.STREAMING} has not been implemented yet.")
        else:
            self.transport = Poller(context, events)

    def start(self):
        self.transport.start()

    def stop(self):
        self.transport.stop()